libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts

```diff
@@ -33,15 +34,25 @@ export class SegmentedControlComponent implements OnChanges {
 
   activeSegment: SegmentItem;
 
+  private isInitializing = false;
+
   onSegmentSelect(item: SegmentItem) {
-    this.activeSegment = item;
-    this.items.forEach((segment) => (segment.checked = this.activeSegment.id === segment.id));
-    this.segmentSelect.emit(this.activeSegment);
+    if (!this.isInitializing) {
+      this.activeSegment = item;
+      this.segmentSelect.emit(this.activeSegment);
+    }
   }
 
   ngOnChanges(changes: SimpleChanges): void {
     if (changes.items) {
-      this.activeSegment = this.items.find((item) => item.checked);
+      // Flag to prevent emitting onSegmentSelect event if previous items exists
+      // Is cleared in ngAfterViewChecked
+      this.isInitializing = true;
+      this.activeSegment = (this.items || []).find((item) => item.checked);
     }
   }
+
+  ngAfterViewChecked(): void {
+    this.isInitializing = false;
+  }
```

Comment:
Can't this be written more concisely as a single `ngOnInit`, since it's run once after the first `ngOnChanges`? That way, you can eliminate the logic of "book keeping" with the `isInitializing` variable.

Reply:
Nope, you have to listen to changes on the items property (like rebinding with "same" items but an updated badge)

Reply:
Ah yeah... missed that part, disregard this comment this :)